src/software/proto/primitive/primitive_msg_factory.h

```diff
@@ -24,13 +24,35 @@ struct AutoChipOrKick
         double autokick_speed_m_per_s;
         double autochip_distance_m;
     };
+
+    bool operator==(const AutoChipOrKick &other) const
+    {
+        if (auto_chip_kick_mode == other.auto_chip_kick_mode)
+        {
+            if (auto_chip_kick_mode != AutoChipOrKickMode::OFF)
+            {
+                return autokick_speed_m_per_s == other.autokick_speed_m_per_s;
+            }
+            else
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    bool operator!=(const AutoChipOrKick &other) const
+    {
+        return !((*this) == other);
+    }
 };
 
 /**
  * PHYSICAL_LIMIT maximum speed allowed by the physical limits of the robot
  * STOP_COMMAND maximum speed allowed when responding to a stop command
+ * TIPTOE maximum speed allowed does not bump the ball to kick speed
```

Comment:
I feel like there's a word missing b/w "allowed" and "does".

Reply:
good catch